### DISCUSSION TITLE
How to use regex when fetching patch dependencies?

### DISCUSSION THREAD

**Discussion by afontan** [renovatebot/renovate#18421] [Request Help] [answered]
Hi, I'm struggling with a problem, sometimes a private dependency contains a `dev` word in the name, which means that we shouldn't migrate to it, but renovate is considering that version as a candidate to update.

E.g

Last version: `mylib:1.0.1-dev`

I need to avoid versions with the word dev in its name, I read the documentation but still don't get on how to use regex to get/discard specific versions, could someone help me here?

**Comment by rarkins** [comment, marked as answer]
You can use a package rule with matchPackageNames and allowedVersions. But usually Renovate should filter out development versions by default. What manager/datasource is this?

**Reply by afontan** [reply]
so defining a regex inside allowedVersions will do the work?

currently, I'm using only gradle. The current json is:

```
{
    "enabledManagers": [
        "gradle"
    ],
    "labels": [
        "dependencies"
    ],
    "extends": [
        "config:base",
        "group:allNonMajor",
        ":combinePatchMinorReleases",
        ":automergePatch",
        ":prConcurrentLimit10",
        ":prHourlyLimitNone",
        ":ignoreUnstable",
        ":noUnscheduledUpdates",
        ":renovatePrefix",
        ":semanticCommits",
        ":rebaseStalePrs",
        ":enableVulnerabilityAlerts"
    ],
    "prCreation": "not-pending",
    "schedule": [
        "before 9am every weekday"
    ],
    "commitMessagePrefix": "[Bump] "
}
```

**Reply by rarkins** [reply]
Yes. Search this repo for allowedVersions and you should see some examples from past discussions. I'm on mobile right now so can't easily link you directly

**Reply by afontan** [reply]
~~It worked, thanks for your help!~~
Here's the solution in the case that is useful for others

[UPDATED] Thanks @viceice 
```
{
    "enabledManagers": [
        "gradle"
    ],
    "labels": [
        "dependencies"
    ],
    "packageRules": [
        {
            "matchPackagePatterns":["*"],
            "allowedVersions": "/^((?!dev).)*$/"
        }
    ],
    "extends": [
        "config:base",
        "group:allNonMajor",
        ":combinePatchMinorReleases",
        ":prConcurrentLimit10",
        ":prHourlyLimitNone",
        ":ignoreUnstable",
        ":noUnscheduledUpdates",
        ":renovatePrefix",
        ":semanticCommits",
        ":rebaseStalePrs",
        ":enableVulnerabilityAlerts"
    ],
    "prCreation": "not-pending",
    "schedule": [
        "before 9am every weekday"
    ],
    "commitMessagePrefix": "[Bump] "
}
```

**Reply by viceice** [reply]
your package rule is wrong, it does nothing, because a `matchXXX` field is missing.

**Reply by afontan** [reply]
~~but seems to be working, how could that be?~~ what should I put in the match field?

**Reply by viceice** [reply]
something like this `matchPackagePatterns:['*']`[^1] or `matchPackageNames:['mylib']`[^2]

[^1]: https://docs.renovatebot.com/configuration-options/#matchpackagepatterns
[^2]: https://docs.renovatebot.com/configuration-options/#matchpackagenames

**Reply by afontan** [reply]
I just updated the comment, thanks @viceice!
